test/provision/Makefile

```diff
@@ -16,7 +16,7 @@ endif
 
 .PHONY: provision
 provision: terraform/backend
-	$(MAKE) ansible/dependencies ANSIBLE_FOLDER=$(ANSIBLE_FOLDER)
+	$(MAKE) ansible/dependencies ANSIBLE_FOLDER=$(ANSIBLE_FOLDER) ANSIBLE_ENV_SCRIPT=$(ANSIBLE_FOLDER)/set_ansible_env.sh
```

Comment:
What do you think running the `ssh-add "$CROWDSTRIKE_ROLE_PULL_KEY"` command before doing the "make ansible/dependencies" in this Makefile so we make CROWDSTRIKE_ROLE_PULL_KEY only required in the provision environment (Makefile)?

Reply:
Thus we could remove the ANSIBLE_ENV_SCRIPT and the variable exports.

Reply:
I guess I can run `eval $$(ssh-agent -s) && ssh-add ...` in the same line, let me try and provision some canaries with that

Reply:
Ah! So the issue is that the ssh-agent is not available in the Ansible dependencies shell? What about exposing a generic SSH key variable from the Ansible common Makefile and do the ssh-add if defined? I think it would be great if we only define and check the CrowdStrike variable when needed (provision)

Reply:
Yup, I ended up with something similar, WDYT? Let me know if you see better ways to achieve it!